Source/WebCore/html/LinkRelAttribute.cpp

```diff
@@ -40,67 +40,85 @@
 
 namespace WebCore {
 
+using LinkRelAttributeHandler = void (*)(LinkRelAttribute& relAttribute);
+using LinkRelAttributeMap = HashMap<ASCIILiteral, LinkRelAttributeHandler>;
+static LinkRelAttributeMap createLinkRelAttributeMap()
+{
+    static NeverDestroyed<LinkRelAttributeMap> linkRelAttributeMap = LinkRelAttributeMap({
```

Comment:
Since there are so few items, binary search is probably faster than hash table lookup.

Reply:
See isPotentialCustomElementNameCharacter in Document.cpp for an example of binary search.

Reply:
There is a single place where this can be implemented now (findLinkType), but it's not done as I wasn't quite sure about the best approach. I did look at that method and also did some research into writing custom a compare method as would be needed here, but it seems quite involved and I couldn't easily find precedent in the codebase.

I'm kinda hoping that since the performance should be about equivalent with what we have today we can take this patch as-is and separately figure out if we want to add a binary search later on.